Specs compliance documentation (#100)
assemblee-virtuelle/activitypods: website/src/pages/specs/solid.md

---
layout: ~/layouts/MdLayout.astro
title: Solid conformance
---

Solid is a set of specifications.

### WebID

> All ActivityPub actors are also WebID.

Reviewer: ```suggestion
All ActivityPub actors are also WebIDs.
```

### Interoperability

[Solid Application Interoperability](https://solid.github.io/data-interoperability-panel/specification) is a complex specification which is not yet finished. Since we don't use [Solid-OIDC](#solid-oidc), we use ActivityPub and a new `Register` activity for app registration (applications are regular ActivityPub actors). We have also decided not to use ShapeTrees, which add nothing to our architecture: instead, applications only declare the classes they want to manage. For the rest, we chose to implement only the parts of the spec that we needed.

What we implement:

- Applications declare their needs through `AccessNeedGroups` and `AccessNeeds`, which are displayed on a user-readable screen.
- Before registration, `AccessGrants` and `DataGrants` are created on the user's Pod, and sent along with the `Register` activity.
- When the user receives an `Accept` activity for the registration, an `ApplicationRegistration` is saved on the user's Pod.
- Whenever the user connects again to the application, we check that the `AccessNeeds` haven't changed. If needed, we show an authorization screen again and `AccessGrants` are updated.

What we don't implement:

> - Authorization flow is done ActivityPub, with a simple `Register` activity.

Reviewer: ```suggestion
- Authorization flow is handled using ActivityPub, with a simple `Register` activity and not with solid (yet).
```

> - We only handle registration with `Applications`, not with other `SocialAgents` (we continue to use WAC permissions for the later)

Reviewer: ```suggestion
- We only handle registration with `Applications`, not with other `SocialAgents` (we continue to use WAC permissions for the latter)
```
